src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1451,6 +1452,23 @@ private void CheckMemberNamesDistinctFromType(DiagnosticBag diagnostics)
             }
         }
 
+        private void CheckRecordMemberNames(DiagnosticBag diagnostics)
+        {
+            if (declaration.Kind != DeclarationKind.Record)
+            {
+                return;
+            }
+
+            Dictionary<string, ImmutableArray<Symbol>> membersByName = GetMembersByName();
```

Comment:
>GetMembersByName [](start = 71, length = 16)

Could we simply use GetMembers("Clone") and then simply iterate through the result?